doc/SonarQube-Integration.md

```diff
@@ -0,0 +1,167 @@
+dashboardUrl=http://localhost:9000/dashboard?id=com.tom%3Asonarqube-jacoco-code-coverage
+ceTaskId=AX0vnvr4_QGKX8b7Yz_v
+ceTaskUrl=http://localhost:9000/api/ce/task?id=AX0vnvr4_QGKX8b7Yz_v
+```
+
+The information extracted fron this file is used to query about SQ issues, measures and task status.
+
+# References
+1. [CloudBees Video on SonarQube integration with Jenkins](https://www.youtube.com/watch?v=KsTMy0920go)
+2. SonarQube WEB API: https://sonarcloud.io/web_api/
+
+## Verification 
+The plugin is using the following API calls 
+1. [Issues Search](https://sonarcloud.io/web_api/api/issues/search)
+2. [Measures Search Histhory](https://sonarcloud.io/web_api/api/measures/search_history)
+3. [Task CE status](https://sonarcloud.io/web_api/api/ce/task)
+
+After seetting up a security token in SonarQube use the following curl calls to verify manually the API integration
+with SonarQube.
+
+### Measures Search Histhory
+export TOKEN=****************
+curl -s -G -u ${TOKEN}: \
+--data-urlencode "componentKey=com.tom:sonarqube-jacoco-code-coverage" \
+--data-urlencode "component=com.tom:sonarqube-jacoco-code-coverage" \
+--data-urlencode "metricKeys=ncloc" \
+http://localhost:9000/api/measures/component | jq .
+```
+{
+  "component": {
+    "key": "com.tom:sonarqube-jacoco-code-coverage",
+    "name": "sonarqube-jacoco-code-coverage",
+    "qualifier": "TRK",
+    "measures": [
+      {
+        "metric": "ncloc",
+        "value": "9"
+      }
+    ]
+  }
+}
+```
+
+
+### Issues Search
+export TOKEN=****************
+curl -s -G -u ${TOKEN}: \
+--data-urlencode "componentKeys=com.tom:sonarqube-jacoco-code-coverage" \
+--data-urlencode "resolved=false" \
+--data-urlencode "severities=INFO" \
+http://localhost:9000/api/issues/search?ps=1 | jq .
+
+```
+{
+  "total": 0,
+  "p": 1,
+  "ps": 1,
+  "paging": {
+    "pageIndex": 1,
+    "pageSize": 1,
+    "total": 0
+  },
+  "effortTotal": 0,
+  "issues": [],
+  "components": [],
+  "facets": []
+}
+```
+ 
+ 
+### Task status
+export TOKEN=****************
+curl -s -G -u ${TOKEN}: \
+--data-urlencode "id=com.tom:sonarqube-jacoco-code-coverage" \
+http://localhost:9000/api/ce/task?id=AX0vnvr4_QGKX8b7Yz_v | jq .
+```
+{
+  "task": {
+    "id": "AX0vnvr4_QGKX8b7Yz_v",
+    "type": "REPORT",
+    "componentId": "AX0m-2Bde0ytHUET2DEl",
+    "componentKey": "com.tom:sonarqube-jacoco-code-coverage",
+    "componentName": "sonarqube-jacoco-code-coverage",
+    "componentQualifier": "TRK",
+    "analysisId": "AX0vnv7gkMePnxoN1IDV",
+    "status": "SUCCESS",
+    "submittedAt": "2021-11-17T20:38:07+0000",
+    "submitterLogin": "admin",
+    "startedAt": "2021-11-17T20:38:08+0000",
+    "executedAt": "2021-11-17T20:38:22+0000",
+    "executionTimeMs": 13679,
+    "hasScannerContext": true,
+    "warningCount": 0,
+    "warnings": []
+  }
+}
+```
+
+## Development
+### Build
+mvn clean package hpi:hpi
+
+### Deploy
+Use the Advanced tab in `Manage Plugins` page to upload `target/influxdb.hpi` file. Restart Jenkins.
```

Comment:
This has nothing to do with SonarQube integration, so it should not be part of this document. 99% of users are interested in using the plugin, not developing it. The only relevant part of this section is
```
withSonarQubeEnv('SonarQube') {
    influxDbPublisher(selectedTarget: 'influxdb_v2',)
}
```
which specifies that you need to run `InfluxDbPublisher` in the same `withSonarQubeEnv` block as you ran SonarQube analysis.